### PR TITLE
Lazy index for fixed length windower

### PR DESCRIPTION
**Pull request by PierreGtch**
This adds the option to use lazily compute the trial bounds when using `create_fixed_length_windows`.

Useful when working with small strides that can lead to metadata DataFrames nearly the same size as the dataset itself.

### CODE REVIEW

**Comment by bruAristimunha** [comment]
@robintibor, can you review this modification on temple university dataset?

**Review by bruAristimunha — CHANGES_REQUESTED** [review]
very small comments, but lgtm

File: braindecode/preprocessing/windowers.py [hunk 1]
```diff
@@ -253,6 +376,9 @@ def create_fixed_length_windows(
     flat: dict | None
         Epoch rejection parameters based on flatness of signals. If None, no
         rejection based on flatness is done. See mne.Epochs.
+    lazy_metadata: bool
```
[the remark this answers is not in the document]
Comment (PierreGtch) [reply braindecode/preprocessing/windowers.py:395]: This argument should only be a bool, not None

**Comment by PierreGtch** [comment]
Thanks @bruAristimunha for the comments! :)

**Review by bruAristimunha — APPROVED** [review]

**Comment by bruAristimunha** [comment]
Thank you @PierreGtch =)

**Pull request merged**